base/runtime/core_builtin_soa.odin

```diff
@@ -425,4 +425,101 @@ clear_soa_dynamic_array :: proc(array: ^$T/#soa[dynamic]$E) {
 @builtin
 clear_soa :: proc{
 	clear_soa_dynamic_array,
-}
\ No newline at end of file
+}
+
+// Converts soa slice into a soa dynamic array without cloning or allocating memory
+@(builtin, require_results)
+into_dynamic_soa :: proc(array: $T/#soa[]$E) -> #soa[dynamic]E {
+	d: #soa[dynamic]E
+	footer := raw_soa_footer_dynamic_array(&d)
+	footer^ = {
+		cap = len(array),
+		len = 0,
+		allocator = nil_allocator(),
+	}
+
+	field_count: uintptr
+	when intrinsics.type_is_array(E) {
+		field_count = len(E)
+	} else {
+		field_count = uintptr(intrinsics.type_struct_field_count(E))
+	}
+
+	array := array
+	dynamic_data := uintptr(&d)
+	slice_data := uintptr(&array)
+	for _ in 0..<field_count {
+		(^uintptr)(dynamic_data)^ = (^uintptr)(slice_data)^
+		dynamic_data += size_of(rawptr)
+		slice_data += size_of(rawptr)
+	}
```

Comment:
This could have been written like this:
```odin
dynamic_data := ([^]rawptr)(&d)[:field_count]
slice_data   := ([^]rawptr)(&array)[:field_count]
copy(dynamic_data, slice_data)
```

Reply:
I like that better too. I'll update it to that in a moment.